### ISSUE TITLE
Different results from visual studio and leetcode editor

### ISSUE THREAD

**Issue by Dl2kten** [LeetCode-Feedback/LeetCode-Feedback#10097]
#### Category of the bug
- [ ] Question
- [ This one] Solution
- [ ] Language
- [ ] Missing Test Cases 

#### Description of the bug

I'm getting different results from my code copied from Visual Studio to leetcode

#### Code you used for Submit/Run operation

```
// Task Scheduler
class Solution {
public:
    void sort_value(std::vector <std::pair<char, int>>& letters, 
        std::unordered_map<char, int>& letter_map) {
        for (auto& i : letter_map) {
            letters.push_back(i);
        }

        std::sort(letters.begin(), letters.end(), [](std::pair<char, int> a, std::pair<char, int> b) {
            return a.second > b.second;
        });
        letter_map.clear();

        for (const auto& i : letters) {
            letter_map.insert(i);
        }
    }

    int leastInterval(std::vector<char>& tasks, int n) {
        std::vector <std::pair<char, int>> letters;
        std::unordered_map<char, int> letter_map;
        int idle_count = 0;

        for (const auto& task : tasks) {
            letter_map[task]++;
        }

        sort_value(letters, letter_map);

        auto count = 0, from_start = 0;
        auto beg = letter_map.begin();

        for (std::size_t i = 0; i < tasks.size(); ++i) {
            if (count++ != n + 1) {
                if (beg != letter_map.end()) {
                    beg->second--;
                    if (beg->second == 0) {
                        letter_map.erase(beg->first);
                        beg = letter_map.begin();
                        int j = 0;
                        while (j++ != from_start) {
                            ++beg;
                        }
                        continue;
                    }

                    ++beg;
                    ++from_start;
                }
                else {
                    --i;
                    ++idle_count;
                }
                
            }
            else {
                beg = letter_map.begin();
                count = 0;
                from_start = 0;
                --i;
            }
        }

        return tasks.size() + idle_count;
    }
};
```

#### Expected behavior

I expected to get 16 for test case 3 but got 22

#### Screenshots

![image](https://user-images.githubusercontent.com/22644604/204958827-f78b1689-0803-43b7-a0ae-2283ad748bdc.png)
![image](https://user-images.githubusercontent.com/22644604/204958890-7be984ab-92c1-439c-a90e-b6571342a974.png)

**Comment by LC-Pam** [comment]
Thank you for reaching out to us. To help us further assess this issue and reproduce this issue on our end, would you please share the following information with us:

The  language you submitted to reach this result.
The problem name or page.

We appreciate your time and support. If you have any other questions, please don't hesitate to let us know.

**Comment by Dl2kten** [comment]
I used C++, the problem name is Task Scheduler.

**Comment by LC-Pam** [comment]
Thank you for reaching out to us, the judger appears to be working correctly, please feel free to refer to the discuss section of the problem to see how other users' have optimized their codes!
When you look at the different outcomes of "Run Code" and "Submit" for this test case, you can see the same input and expected value but different output value.

We've documented the cause and solutions of this kind of issue here: https://support.leetcode.com/hc/en-us/articles/360011834174-I-encountered-Wrong-Answer-Runtime-Error-for-a-specific-test-case-When-I-test-my-code-using-this-test-case-it-produced-the-correct-output-Why-

**Comment by Dl2kten** [comment]
Yeah, but my issue is that the judger is giving different results from Visual Studio.  I looked at the article, there are no globals in the code I submitted and if index out of range that would be a runtime error which is not the case here.  I've debugged the code when I ran it in visual studio, so if there's a logic issue that would be my miss but I am getting the correct answer in Visual Studio.  I've looked at other submissions and changed my code.  Just letting you guys know there might be an issue with the judger.
